### PR TITLE
HPCC-32420 Add client and URL instrumentation to the ESP server span

### PR DESCRIPTION
**Pull request by timothyklemm**
- Add client.address attribute using incoming peer IP address.
- Add url.full using incoming values, redacting unknown query parameters that may contain sensitive data. If redaction occurs, a synthesized parameter will be included to indicate how many parameters have been omitted.
- Add the url.query.redacted attribute when the number of parameters redacted from url.full is not zero and the trace level is at least traceDetailed.

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by rpastrana — COMMENTED** [review]
@timothyklemm looks good, just one request.

File: esp/bindings/http/platform/httpservice.cpp [hunk 1]
```diff
@@ -209,6 +209,94 @@ void checkSetCORSAllowOrigin(EspHttpBinding *binding, CHttpRequest *req, CHttpRe
...
+{
+    // General Notice: Span attributes added here are recorded as shown. Certain backend collectors
+    // may transform the names seen here. For example, `client.address` may become
+    // `labels.client_address` on the backend. Not all attributes are affected by this.
+    if (!queryTraceManager().isTracingEnabled())
+        return;
+    ISpan* span = queryThreadedActiveSpan();
+    if (!span->isRecording())
+        return;
+    StringBuffer peer, peerPort;
```
Comment (rpastrana) [inline esp/bindings/http/platform/httpservice.cpp:222]: peerport no longer in use.

**Review by rpastrana — APPROVED** [review]
@timothyklemm One minor change needed. Squash after fixing.

**Comment by timothyklemm** [comment]
@ghalliday please merge

File: esp/bindings/http/platform/httpservice.cpp [hunk 2]
```diff
@@ -209,6 +209,94 @@ void checkSetCORSAllowOrigin(EspHttpBinding *binding, CHttpRequest *req, CHttpRe
...
+                        redacted.append("&");
+                    redacted.append(name);
+                }
+            }
+            else
+            {
+                if (!query.isEmpty())
+                    query.append('&');
+                query.append(name);
+                const char* value = parameters->queryProp(name);
```
Comment (ghalliday) [inline esp/bindings/http/platform/httpservice.cpp:273]: minor: use it->queryPropValue() rather than re-resolving the entry.

**Review by ghalliday — COMMENTED** [review]
@timothyklemm looks good.  One minor comment.  I could merge as is, but probably worth changing.

**Comment by timothyklemm** [comment]
@ghalliday I've made the requested change and squashed.

**Review by ghalliday — APPROVED** [review]
Looks good.  Please squash once Rodrigo has approved.

**Review by rpastrana — APPROVED** [review]

**Comment by timothyklemm** [comment]
@ghalliday this is ready to merge

**Comment by ghalliday** [comment]
@timothyklemm thanks.  It is better if you don't rebase as well as squash - it makes it hard to check the squash did not introduce any other issues.

**Comment by timothyklemm** [comment]
> @timothyklemm thanks. It is better if you don't rebase as well as squash - it makes it hard to check the squash did not introduce any other issues.

Sorry. It was not intentional. I updated my candidate branch to work on a different issue. It didn't cross my mind that squashing this would also change where this branched from.

**Pull request merged**